fdbserver/storageserver.actor.cpp

```diff
@@ -9150,6 +9163,24 @@ ACTOR Future<Void> storageServerCore(StorageServer* self, StorageServerInterface
 			when(FetchCheckpointRequest req = waitNext(ssi.fetchCheckpoint.getFuture())) {
 				self->actors.add(fetchCheckpointQ(self, req));
 			}
+			when(UpdateCommitCostRequest req = waitNext(ssi.updateCommitCostRequest.getFuture())) {
+				// In case we received an old request/duplicate request, due to, e.g. network problem
+				if (req.postTime < self->busiestWriteTagContext.lastUpdateTime) {
```

Comment:
One other possible issue here is if we switch ratekeepers, in which case the clocks may not be in sync. An easy though imperfect solution to this could be to include an RK ID and log the event and update the time unconditionally any time the ID changes. That's probably good enough, though if desired there are other options that could be more accurate.

Reply:
Good point, I suppose this would not happen often, maybe take it as an improvement later.

Reply:
I think this is pretty rare, but as this is indeed a simple one, I added the Ratekeeper ID.